samplers/parser.go

```diff
@@ -433,44 +422,53 @@ func ParseEvent(packet []byte) (*UDPEvent, error) {
 			if foundHostname {
 				return nil, errors.New("Invalid event packet, multiple hostname sections")
 			}
-			ret.Hostname = string(pipeSplitter.Chunk()[2:])
+			ret.Tags[dogstatsd.EventHostnameTagKey] = string(pipeSplitter.Chunk()[2:])
 			foundHostname = true
 		case bytes.HasPrefix(pipeSplitter.Chunk(), []byte{'k', ':'}):
 			if foundAggregation == true {
 				return nil, errors.New("Invalid event packet, multiple aggregation key sections")
 			}
-			ret.Aggregation = string(pipeSplitter.Chunk()[2:])
+			ret.Tags[dogstatsd.EventAggregationKeyTagKey] = string(pipeSplitter.Chunk()[2:])
 			foundAggregation = true
 		case bytes.HasPrefix(pipeSplitter.Chunk(), []byte{'p', ':'}):
 			if foundPriority == true {
 				return nil, errors.New("Invalid event packet, multiple priority sections")
 			}
-			ret.Priority = string(pipeSplitter.Chunk()[2:])
-			if ret.Priority != "normal" && ret.Priority != "low" {
+			ret.Tags[dogstatsd.EventPriorityTagKey] = string(pipeSplitter.Chunk()[2:])
+			if ret.Tags[dogstatsd.EventPriorityTagKey] != "normal" && ret.Tags[dogstatsd.EventPriorityTagKey] != "low" {
 				return nil, errors.New("Invalid event packet, priority must be normal or low")
 			}
 			foundPriority = true
 		case bytes.HasPrefix(pipeSplitter.Chunk(), []byte{'s', ':'}):
 			if foundSource == true {
 				return nil, errors.New("Invalid event packet, multiple source sections")
 			}
-			ret.Source = string(pipeSplitter.Chunk()[2:])
+			ret.Tags[dogstatsd.EventSourceTypeTagKey] = string(pipeSplitter.Chunk()[2:])
 			foundSource = true
 		case bytes.HasPrefix(pipeSplitter.Chunk(), []byte{'t', ':'}):
 			if foundAlert == true {
 				return nil, errors.New("Invalid event packet, multiple alert sections")
 			}
-			ret.AlertLevel = string(pipeSplitter.Chunk()[2:])
-			if ret.AlertLevel != "error" && ret.AlertLevel != "warning" && ret.AlertLevel != "info" && ret.AlertLevel != "success" {
+			aType := string(pipeSplitter.Chunk()[2:])
+			if aType != "error" &&
+				aType != "warning" &&
+				aType != "info" &&
+				aType != "success" {
 				return nil, errors.New("Invalid event packet, alert level must be error, warning, info or success")
 			}
+			ret.Tags[dogstatsd.EventAlertTypeTagKey] = aType
 			foundAlert = true
 		case pipeSplitter.Chunk()[0] == '#':
-			if ret.Tags != nil {
+			if foundTags == true {
 				return nil, errors.New("Invalid event packet, multiple tag sections")
 			}
 			tags := strings.Split(string(pipeSplitter.Chunk()[1:]), ",")
-			ret.Tags = tags // no need to sort, we don't aggregate on this
+			mappedTags := ParseTagSliceToMap(tags)
+			// We've already added some tags, so we'll just add these to the ones we've got.
+			for k, v := range mappedTags {
+				ret.Tags[k] = v
```

Comment:
I wonder if we should check that we're not overwriting any dogstatsd tags with user-specified tags that match the names veneur uses internally.